scripts/pretraining/bert/run_pretraining.py

```diff
@@ -321,10 +340,17 @@ def train(args):
                     ns_pred_list.append(nsp_score)
 
                 running_num_tks += valid_length.sum().as_in_ctx(mx.cpu())
-
-            for loss in loss_l:
-                loss.backward()
-
+            if args.use_amp:
+                with mx.autograd.record():
+                    with amp.scale_loss(loss_l, trainer) as loss_l:
+                        for loss in loss_l:
+                            loss.backward()
+                norm_clip_mult = num_workers * trainer._amp_loss_scaler.loss_scale
```

Comment:
We can directly use `trainer.amp_loss_scale` as in https://github.com/dmlc/gluon-nlp/blob/dfdfb6d12f82e6caa47bbd7430efcd64dfa311dc/scripts/machine_translation/train_transformer.py#L708